modules/network-annotation-processor/src/main/java/org/apache/ignite/network/processor/internal/serialization/MessageDeserializerGenerator.java

```diff
@@ -0,0 +1,154 @@
+    /** */
+    private final ProcessingEnvironment processingEnv;
+
+    /**
+     * Message Types declarations for the current module.
+     */
+    private final MessageGroupWrapper messageGroup;
+
+    /** */
+    public MessageDeserializerGenerator(ProcessingEnvironment processingEnv, MessageGroupWrapper messageGroup) {
+        this.processingEnv = processingEnv;
+        this.messageGroup = messageGroup;
+    }
+
+    /**
+     * Generates a {@link MessageDeserializer} class for the given network message type.
+     */
+    public TypeSpec generateDeserializer(MessageClass message) {
+        processingEnv.getMessager()
+            .printMessage(Diagnostic.Kind.NOTE, "Generating a MessageDeserializer", message.element());
+
+        FieldSpec msgField = FieldSpec.builder(message.builderClassName(), "msg")
+            .addModifiers(Modifier.PRIVATE, Modifier.FINAL)
+            .build();
+
+        return TypeSpec.classBuilder(message.simpleName() + "Deserializer")
+            .addSuperinterface(ParameterizedTypeName.get(ClassName.get(MessageDeserializer.class), message.className()))
+            .addField(msgField)
+            .addMethod(
+                MethodSpec.constructorBuilder()
+                    .addParameter(messageGroup.messageFactoryClassName(), "messageFactory")
+                    .addStatement("this.$N = messageFactory.$L()", msgField, message.asMethodName())
+                    .build()
+            )
+            .addMethod(
+                MethodSpec.methodBuilder("klass")
+                    .addAnnotation(Override.class)
+                    .addModifiers(Modifier.PUBLIC)
+                    .returns(ParameterizedTypeName.get(ClassName.get(Class.class), message.className()))
+                    .addStatement("return $T.class", message.className())
+                    .build()
+            )
+            .addMethod(
+                MethodSpec.methodBuilder("getMessage")
+                    .addAnnotation(Override.class)
+                    .addModifiers(Modifier.PUBLIC)
+                    .returns(message.className())
+                    .addStatement("return $N.build()", msgField)
+                    .build()
+            )
+            .addMethod(readMessageMethod(message, msgField))
+            .addOriginatingElement(message.element())
+            .addOriginatingElement(messageGroup.element())
+            .build();
+    }
+
+    /**
+     * Generates the {@link MessageDeserializer#readMessage(MessageReader)} implementation.
+     */
+    private MethodSpec readMessageMethod(MessageClass message, FieldSpec msgField) {
+        MethodSpec.Builder method = MethodSpec.methodBuilder("readMessage")
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .returns(boolean.class)
+            .addParameter(MessageReader.class, "reader")
+            .addException(MessageMappingException.class);
+
+        List<ExecutableElement> getters = message.getters();
```

Comment:
List is not sorted, but you already know it

Reply:
Nice catch!